src/ejabberd_c2s.erl

```diff
@@ -3195,12 +3194,10 @@ handover_session(SD, From)->
     {stop, {handover_session, From}, SD}.
 
 do_handover_session(SD, UnreadMessages) ->
-    {N, Messages} = flush_messages(UnreadMessages),
-    NewSize = N + SD#state.stream_mgmt_buffer_size,
-    NewBuffer = Messages ++ SD#state.stream_mgmt_buffer,
+    Messages = flush_messages(UnreadMessages),
+    NewCsiBuffer = Messages ++ SD#state.csi_buffer,
```

Comment:
Why these are added to CSI buffer now, instead of SM?

Reply:
actually it is wrong to add it to the `stream_mgmt_buffer`, there are action performed on the messages before they land in `stream_mgmt_buffer`